docs/design/coreclr/botr/logging.md

```diff
@@ -0,0 +1,70 @@
+
+*NOTE: This document will repeatedly reference many environment variables. For a detailed list of environment variables supported by the runtime and explanations of what they do, consult [`clrconfigvalues.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/clrconfigvalues.h), [`gcconfig.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/gc/gcconfig.h), and [`jitconfigvalues.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/jitconfigvalues.h).*
+
+Types of logging
+================
+
+ETW / EventPipe logging
+-----------------------
+"Event Tracing for Windows", or ETW, is an extensible tracing system built into Windows that various applications and services can use to report events in a way that allows the end user to filter and record them dynamically. ETW is consumed by tools like perfview, xperf, UIforETW, or WPR/WPA. This system is frequently used by customers to understand what is happening in development or production environments.
+
+EventPipe is a cross-platform tracing system similar to ETW that is used widely in the .NET runtime.
+For a detailed overview of EventPipe, see [the EventPipe documentation pages on .NET Learn](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/eventpipe). It offers excellent performance along with lots of flexibility, so it is one of the main ways to get information from an active runtime that we advertise. Most events that we support in debug builds also are exposed in release builds.
+
+For basic scenarios, you can use the [`dotnet-counters`](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-counters) tool to monitor events and performance counters the runtime reports through EventPipe. For example, to collect the default counters from a project while running it, you could use `dotnet-counters collect -- dotnet run`.
+
+You can also use the [`dotnet-trace`](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-trace) tool to capture EventPipe events from the runtime on the fly, either saving them to a file or sending them to your console. It provides command line options you can use to filter events by severity level or based on keywords. For example, to capture high-severity GC events only from a project while running it, you could use `dotnet-trace collect --clreventlevel error --clrevents gc -- dotnet run`.
+
+Please see the linked documentation pages for more information on how to use each of the above tools. They have an extensive set of options and some helpful features like time limits, attaching to existing processes, and configurable output formatting.
+
+For advanced scenarios, on Windows you can also use standard ETW tools like [WPR](https://learn.microsoft.com/en-us/windows-hardware/test/wpt/windows-performance-recorder) to capture events from EventPipe, and on Linux you can use [LTTNG](https://lttng.org/).
```

Comment:
```suggestion
For advanced scenarios, on Windows you can also use standard ETW tools like [WPR](https://learn.microsoft.com/en-us/windows-hardware/test/wpt/windows-performance-recorder), and on Linux you can use [LTTNG](https://lttng.org/).
```
I think WPR is ETW only. I do not think that it works with EventPipe.

Reply:
I must have misunderstood previous comments, I thought eventpipe funnels events to ETW as well? Is that only EventSource, even though the eventpipe API in C++ is in the ETW:: namespace?

Reply:
EventPipe is first and foremost a cross-process communication protocol.

[EventPipe native component](https://github.com/dotnet/runtime/tree/main/src/native/eventpipe) implements the provider side of the protocol. (The listener side of the protocol is in https://github.com/microsoft/perfview/tree/main/src/TraceEvent library.) 

The multicasting over EventPipe/ETW/LTTNG/custom is done in EventSource (for events originating in managed code) and C/C++ macros (for events originating in native code). A few examples:

- https://github.com/dotnet/runtime/blob/ac9407554954d920a15337f8af471b1d4f02a506/src/coreclr/inc/eventtracebase.h#L100 is example of multiplexing on Windows. This code says "return true if ETW or EventPipe is subscribed to this event"
- https://github.com/dotnet/runtime/blob/ac9407554954d920a15337f8af471b1d4f02a506/src/coreclr/inc/eventtracebase.h#L123-L124 is example of multiplexing on non-Windows. This code says "return true if this event EventPipe or LTTNG is subscribed to this event"
- And here is similar logic for EventSource: https://github.com/dotnet/runtime/blob/ac9407554954d920a15337f8af471b1d4f02a506/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs#L2421-L2425

> ETW:: namespace

A lot of tracing-related code is in ETW namespace for historic reasons, but it does not mean that it is ETW specific.

Reply:
I see, that makes a lot of sense. I'll revise this whole portion of it to more clearly separate EventPipe from ETW and EventSource conceptually.

Reply:
I've verified that WPR works with the .NET runtime (and even has a scenario preset for it) so I'm going to use that as the one example tool to call out in this documentation, alongside perfcollect for linux. I'm moving the ETW/LTTNG stuff into a separate section to clearly separate it from EventPipe.